### PR TITLE
(#167) Add support multiple sites in site filter.

### PR DESCRIPTION
**Pull request by blairlearn**

### CODE REVIEW

**Review by adriancofie — CHANGES_REQUESTED** [review]
https://github.com/NCIOCPL/sitewide-search-app/pull/171/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R115

Using .constructor isn't usually the first approach for checking types in JS. It works but if you're going this route I believe you should also verify that the variable isn't null or undefined.

I believe it also checks solely for a String object but omits checking the string primitive.

I would lean toward something like:

```if (typeof searchSiteFilter === 'string' || searchSiteFilter instanceof String)```

If you want to be consistent and follow the same approach for the Array, `typeof` will return an object type, so you will need to use 'instanceof`:

```if (searchSiteFilter instanceof Array)```

Lastly,. you can also use the Array static method `Array.isArray(searchSiteFilter)`  which I believe is the preferred methodology these days, but you'll sacrifice the consistency with the string/String approach

@arcepaul  Feel free to weigh in also

**Comment by arcepaul** [comment]
I do like @adrian 's suggestion for the conditional for the sake of consistency.

**Comment by blairlearn** [comment]
@adriancofie Interesting.  I wasn't aware of the `instanceof` operator.  (The use of `.constructor` was a workaround for `typeof` returning `Object` on the array.)

**Review by seyilonge-nci — APPROVED** [review]

**Review by adriancofie — APPROVED** [review]

File: cypress/e2e/common/index.js [hunk 1]
```diff
@@ -43,6 +43,12 @@ Given('{string} is set to {string}', (key, param) => {
 	cy.on('window:before:load', (win) => {
 		if (param === 'null') {
 			win.INT_TEST_APP_PARAMS[key] = null;
+		} else if (key === 'searchSiteFilter') {
+			if (param.includes(','))
```
[the remark this answers is not in the document]
Comment (blairlearn) [reply cypress/e2e/common/index.js]: Curly braces added; also on the `else` statements.

**Review by KateMashkinaNIH — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**